### PR TITLE
Added a change log

### LINKED ISSUE #108: phploc has no change log
I'm currently using phploc 2.0.6 and saw 2.1.0 was released.
Eager to know what's new I was looking for a change log or release notes.
Neither exist.
In the end I browsed through the commit messages between 2.1.0 <=> 2.0.6.
It would be more convenient for implementing developers if there was a change log of some sorts.

That's why I propopse to include one, like explained here:
http://keepachangelog.com/

### PR DESCRIPTION
**Pull request by ravage84**
Closes #108

Browsed through the tag diffs and tried to extract some relevant changes.
Certainly there are many other and probably more relevant changes than I've found.

But let's start with this.

I wasn't sure about the statement that phploc adheres to SemVer, though.
May be it does only "since version x.y.z"?

I didn't pushed the URL references down, since I think it's a bad practise.
Especially when files grow in length (as change logs do very quickly).
This way it is clean and clean.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+# Change Log for phploc
```
Comment (GrahamCampbell) [inline CHANGELOG.md]: PHPLOC

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+This project adheres to [Semantic Versioning](http://semver.org/).
+
+## [Unreleased][unreleased]
+[unreleased]: https://github.com/sebastianbergmann/phploc/compare/2.1.3...HEAD
+
+### Changed
+- Drop support for PHP < 5.4
+
+### Fixed
+- Make installable on php 7
```
Comment (GrahamCampbell) [inline CHANGELOG.md]: This isn't a "fix" imo.

**Comment by ravage84** [comment]
@sebastianbergmann  I didn't create the file all by hand.
May be these few lines help you to setup the change logs for your other projects:

Export git tags with date:
````
git log --tags --simplify-by-decoration --pretty="format:%ai %d"
````
Tags of phploc
````
2015-06-04 15:18:14 +0200  (tag: 2.1.3, origin/2.1)
2015-05-26 16:03:47 +0200  (tag: 2.1.2)
2015-04-12 14:00:39 +0200  (tag: 2.1.1)
2015-03-10 17:49:43 +0100  (tag: 2.1.0)
2014-06-25 10:11:02 +0200  (tag: 2.0.6, origin/2.0)
2014-04-27 08:47:27 +0200  (tag: 2.0.5)
2013-12-18 09:31:50 +0100  (tag: 2.0.4)
2013-11-05 08:24:36 +0100  (tag: 2.0.3)
2013-09-04 10:30:10 +0200  (tag: 2.0.2)
2013-09-03 06:58:48 +0200  (tag: 2.0.1)
2013-08-28 09:46:34 +0200  (tag: 2.0.0)
2013-08-02 18:23:08 +0200  (tag: 2.0.0-BETA1)
2012-11-10 13:45:44 +0100  (tag: 1.7.4)
2012-11-09 08:47:18 +0100  (tag: 1.7.3)
2012-10-11 22:40:26 +0200  (tag: 1.7.2)
2012-10-10 10:43:48 +0200  (tag: 1.7.1)
2012-10-09 17:06:25 +0200  (tag: 1.7.0)
2011-11-17 11:20:11 +0100  (tag: 1.6.4)
2011-11-16 18:08:46 +0100  (tag: 1.6.3)
2011-10-31 08:43:35 +0100  (tag: 1.6.2)
2011-09-05 08:23:51 +0200  (tag: 1.6.2RC1)
2011-01-28 10:56:13 +0100  (tag: 1.6.1)
2011-01-27 16:15:20 +0100  (tag: 1.6.0)
2010-02-09 10:08:36 +0100  (tag: 1.5.1)
2010-01-03 13:47:40 +0100  (tag: 1.5.0)
2009-11-24 09:35:40 +0100  (tag: 1.4.0)
2009-11-12 07:54:33 +0100  (tag: 1.3.2)
2009-11-06 11:53:17 +0100  (tag: 1.3.1)
2009-11-04 18:42:49 +0100  (tag: 1.3.0)
2009-09-11 10:40:50 +0200  (tag: 1.2.0)
2009-06-03 10:28:31 +0200  (tag: 1.1.1)
2009-06-02 16:56:35 +0200  (tag: 1.1.0)
2009-01-26 18:00:39 +0100  (tag: 1.0.0)
````

Search Regex
````
(\d\d\d\d-\d\d-\d\d) (\d\d:\d\d:\d\d \+\d\d\d\d  \(tag: )(\d\.\d\.\d)((, ).*|!?(.*).*)\)
````

General Replace Regex:
````
## [${3}${6}] - ${1}
[${3}${6}]: https://github.com/vendor/package/compare/X.Y.Z...${3}${6}

````

phploc specific Replace Regex:
````
## [${3}${6}] - ${1}
[${3}${6}]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...${3}${6}

````

Generated CHANGELOG.md for phploc:
````
## [2.1.3] - 2015-06-04
[2.1.3]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.1.3

## [2.1.2] - 2015-05-26
[2.1.2]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.1.2

## [2.1.1] - 2015-04-12
[2.1.1]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.1.1

## [2.1.0] - 2015-03-10
[2.1.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.1.0

## [2.0.6] - 2014-06-25
[2.0.6]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.0.6

## [2.0.5] - 2014-04-27
[2.0.5]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.0.5

## [2.0.4] - 2013-12-18
[2.0.4]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.0.4

## [2.0.3] - 2013-11-05
[2.0.3]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.0.3

## [2.0.2] - 2013-09-04
[2.0.2]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.0.2

## [2.0.1] - 2013-09-03
[2.0.1]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.0.1

## [2.0.0] - 2013-08-28
[2.0.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.0.0

## [2.0.0-BETA1] - 2013-08-02
[2.0.0-BETA1]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...2.0.0-BETA1

## [1.7.4] - 2012-11-10
[1.7.4]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.7.4

## [1.7.3] - 2012-11-09
[1.7.3]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.7.3

## [1.7.2] - 2012-10-11
[1.7.2]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.7.2

## [1.7.1] - 2012-10-10
[1.7.1]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.7.1

## [1.7.0] - 2012-10-09
[1.7.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.7.0

## [1.6.4] - 2011-11-17
[1.6.4]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.6.4

## [1.6.3] - 2011-11-16
[1.6.3]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.6.3

## [1.6.2] - 2011-10-31
[1.6.2]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.6.2

## [1.6.2RC1] - 2011-09-05
[1.6.2RC1]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.6.2RC1

## [1.6.1] - 2011-01-28
[1.6.1]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.6.1

## [1.6.0] - 2011-01-27
[1.6.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.6.0

## [1.5.1] - 2010-02-09
[1.5.1]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.5.1

## [1.5.0] - 2010-01-03
[1.5.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.5.0

## [1.4.0] - 2009-11-24
[1.4.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.4.0

## [1.3.2] - 2009-11-12
[1.3.2]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.3.2

## [1.3.1] - 2009-11-06
[1.3.1]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.3.1

## [1.3.0] - 2009-11-04
[1.3.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.3.0

## [1.2.0] - 2009-09-11
[1.2.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.2.0

## [1.1.1] - 2009-06-03
[1.1.1]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.1.1

## [1.1.0] - 2009-06-02
[1.1.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.1.0

## [1.0.0] - 2009-01-26
[1.0.0]: https://github.com/sebastianbergmann/phploc/compare/X.Y.Z...1.0.0

````

**Comment by GrahamCampbell** [comment]
Maybe it's only worth keeping the changes from 2.x.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
+# Change Log for phploc
+All notable changes to phploc are documented in this file
```
Comment (ravage84) [inline CHANGELOG.md]: OK, then it needs to be ``PHPLOC`` here, too.

**Comment by ravage84** [comment]
@GrahamCampbell 
Capitalized PHPLOC and I disagree about keeping only the changes from 2.x.
It does not hurt to keep the whole history together. That way people can get a full picture in one file.

**Comment by GrahamCampbell** [comment]
My idea with dropping those changes was just because the change log is only partial anyway.

**Comment by ravage84** [comment]
How do we progress on this?

**Pull request closed without merging**